File: data/code_pairs/fortran/1674.f90
module gtk_hl_accelerator
  implicit none
contains
  subroutine hl_gtk_widget_add_accelerator(widget, signal, accel_group, accel_key, accel_mods, accel_flags)
    integer, intent(in) :: widget, accel_group
    character(len=*), intent(in) :: signal, accel_key
    integer, intent(in), optional :: accel_mods, accel_flags
    integer :: ikey, imods, iflags

    ! Default values
    if (present(accel_mods)) then
       imods = accel_mods
    else
       imods = 4  ! Simulated GDK_CONTROL_MASK
    end if

    if (present(accel_flags)) then
       iflags = accel_flags
    else
       iflags = 1  ! Simulated GTK_ACCEL_VISIBLE
    end if

    ! Simulate key value conversion
    ikey = iachar(accel_key)

    ! Simulate adding accelerator
    print *, "Adding accelerator to widget", widget, "with key", accel_key
    print *, "Modifiers:", imods, "Flags:", iflags
  end subroutine hl_gtk_widget_add_accelerator
end module gtk_hl_accelerator

File: data/code_pairs/cpp/1674.cpp
#ifndef GTK_HL_ACCELERATOR_HPP
#define GTK_HL_ACCELERATOR_HPP

#include <iostream>
#include <string>

class GtkHlAccelerator {
public:
    static void hl_gtk_widget_add_accelerator(const std::string& widget, const std::string& signal, const std::string& accel_group, const std::string& accel_key, int accel_mods = 4 /* GDK_CONTROL_MASK */, int accel_flags = 1 /* GTK_ACCEL_VISIBLE */) {
        std::cout << "Adding accelerator to widget: " << widget << std::endl;
        std::cout << "Signal: " << signal << ", Accel Group: " << accel_group << std::endl;
        std::cout << "Accel Key: " << accel_key << ", Mods: " << accel_mods << ", Flags: " << accel_flags << std::endl;
    }
};

#endif // GTK_HL_ACCELERATOR_HPP